### PR TITLE
main.go: update the token file atomically

### PR DESCRIPTION
**Pull request by squat**
To ensure that a file containing a token is updated atomically, we need
to use a temporary file and later move the file to the correct location.

Signed-off-by: Lucas Servén Marín <lserven@gmail.com>

### CODE REVIEW

**Comment by squat** [comment]
cc @observatorium/maintainers

File: main.go [hunk 1]
```diff
@@ -196,11 +202,15 @@ func main() {
 					case !t.Valid():
 						level.Error(logger).Log("msg", "token is invalid", "exp", t.Expiry.String())
 					default:
-						if err := ioutil.WriteFile(cfg.file, []byte(t.AccessToken), 0644); err != nil {
-							level.Error(logger).Log("msg", "failed to write token to disk", "err", err)
-						} else {
-							d = t.Expiry.Sub(time.Now()) - cfg.margin
+						if err := ioutil.WriteFile(cfg.tempFile, []byte(t.AccessToken), 0644); err != nil {
```
Comment (onprem) [inline main.go:205]: `ioutil` is deprecated now, it will continue working but it might be a better idea to use `os.WriteFile` here.

**Review by onprem — APPROVED** [review]
Just a small nit, everything else LGTM.

File: main.go (hunk 1 above)
Comment (squat) [reply main.go:205]: Sounds good 😊

Comment (kakkoyun) [reply main.go:205]: We still haven't upgraded to go1.16 so we're good for now.

**Review by kakkoyun — APPROVED** [review]
🥇

**Pull request merged**